RecoTracker/LSTCore/src/alpaka/TrackCandidate.h

```diff
@@ -223,10 +243,22 @@ namespace ALPAKA_ACCELERATOR_NAMESPACE::lst {
             float phi2 = __H2F(quintuplets.phi()[quintupletIndex]);
             float dEta = alpaka::math::abs(acc, eta1 - eta2);
             float dPhi = cms::alpakatools::deltaPhi(acc, phi1, phi2);
-
             float dR2 = dEta * dEta + dPhi * dPhi;
-            if (dR2 < 1e-3f)
-              pixelSegments.isDup()[pixelArrayIndex] = true;
+            // Cut on pLS-T5 embed distance.
+            if (dR2 < 0.02f) {
+              float d2 = 0.f;
+              CMS_UNROLL_LOOP for (unsigned k = 0; k < Params_pLS::kEmbed; ++k) {
+                float diff = pixelSegments.plsEmbed()[pixelArrayIndex][k] - quintuplets.t5Embed()[quintupletIndex][k];
+                d2 += diff * diff;
+              }
+              // Get pLS eta bin for cut value.
+              float absEta1 = alpaka::math::abs(acc, eta1);
+              uint8_t bin_idx = (absEta1 > 2.5f) ? (dnn::kEtaBins - 1) : static_cast<uint8_t>(absEta1 / dnn::kEtaSize);
+
+              if (alpaka::math::sqrt(acc, d2) < dnn::plsembdnn::kWP[bin_idx]) {
```

Comment:
```suggestion
              if (d2 < dnn::plsembdnn::kWP[bin_idx] * dnn::plsembdnn::kWP[bin_idx]) {
```
better redefine `kWP` though

Reply:
Agreed that kWP should be redefined ideally. Updated to avoid the sqrt for now.